### PR TITLE
Add example: Intercom

### PR DESCRIPTION
**Pull request by Nutlope**
Intercom is an extremely popular chat widget used by 170k+ startups. I helped my dad integrate it into his Next.js app so I thought I'd add it as an example since it was a bit of a struggle trying to follow the React instructions Intercom had. On that note, we should explore integrating/partner with them more.

It didn't just work in Next.js due to the use of `window`, so I created an IntercomProvider that anyone can use to integrate Intercom into their Next.js app as shown in this example.

### CODE REVIEW

**Comment by Nutlope** [comment]
Not sure why this failed – does anyone know?
CleanShot 2022-03-17 at 01 09 24@2x

**Comment by awareness481** [comment]
```ts
    const handleRouteChange = (url) => {
      if (typeof window !== 'undefined') {
        updateIntercom()
      }
    }
```
Since this is defined inside `useEffect` you don't need to check for window. 👍

**Comment by Nutlope** [comment]
> 

@awareness481 Thank you so much! Very helpful, I'm going to test then make the change.

**Comment by Nutlope** [comment]
Moving to drafts since I want to look into using next/script and the onLoad function as Balazs suggested

File: examples/with-intercom/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "next": "12.1.0",
+    "react": "17.0.2",
+    "react-dom": "17.0.2"
+  },
+  "devDependencies": {
+    "eslint": "8.11.0",
+    "eslint-config-next": "12.1.0"
```
Comment (balazsorban44) [inline examples/with-intercom/package.json:17]:
```suggestion
```

Not necessary for this example.

File: examples/with-intercom/package.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "name": "with-intercom",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "next": "12.1.0",
```
Comment (balazsorban44) [inline examples/with-intercom/package.json:11]:
```suggestion
    "next": "latest",
```

File: examples/with-intercom/package.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "with-intercom",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
```
Comment (balazsorban44) [inline examples/with-intercom/package.json:8]:
```suggestion
    "start": "next start"
```

**Comment by leerob** [comment]
@Nutlope let's revisit this with App Router + `next/script` later 😄

**Pull request closed without merging**